base/docs/Docs.jl

```diff
@@ -239,11 +239,26 @@ end
 # =================
 
 """
+    getdoc(x::T) -> String
+
+Return the dynamic docstring associated with object `x`.
+"""
+function getdoc end
+
+"""
     Docs.doc(binding, sig)
 
 Returns all documentation that matches both `binding` and `sig`.
+
+If `getdoc` is defined on the value of the binding, then a dynamic docstring is returned instead of one based on the binding itself. 
 """
 function doc(binding::Binding, sig::Type = Union{})
+    if isdefined(binding.mod, binding.var)
+        var = getfield(binding.mod, binding.var)
+        if method_exists(getdoc, (typeof(var),))
+            return Base.Markdown.parse(getdoc(var))
+        end
+    end
```

Comment:
Can be a bit shorter/clearer as

```diff
 function doc(binding::Binding, sig::Type = Union{})
+    if defined(binding)
+        result = getdoc(resolve(binding), sig)
+        result === nothing || return Markdown.parse(result)
+    end
     results, groups = DocStr[], MultiDoc[]
```

with a default `getdoc(obj, sig) = nothing` method. Passing `sig` through to `getdoc` as well may be useful?